internal/observability/stream.go

```diff
@@ -98,7 +99,11 @@ func (s *streamWrapper) SendMessage(ctx context.Context, msg *transport.StreamMe
 
 func (s *streamWrapper) ReceiveMessage(ctx context.Context) (*transport.StreamMessage, error) {
 	msg, err := s.StreamCloser.ReceiveMessage(ctx)
-	s.call.logStreamEvent(err, _successfulStreamReceive, _errorStreamReceive)
+	// Receiving EOF does not constitute an error for the purposes of metrics and alerts.
+	// This is the only special case.
+	// All other log events treat EOF as an error, including when sending a
+	// message or concluding a handshake.
+	s.call.logStreamEvent(err, err == nil || err == io.EOF, _successfulStreamReceive, _errorStreamReceive)
```

Comment:
It needs to be more consistent in this method with err. Note line 108, it wilkl count EOF as error and I guess this is not right.

Need to set "success" once and reuse it then:


success := err == nil || err == io.EOF
s.call.logStreamEvent(err, success..)

s.edge.receives.Inc()
if success {
...

Reply:
This is a good catch. I will have a follow-up.

Reply:
#1885